tests/vmi_lifecycle_test.go

```diff
@@ -571,6 +573,69 @@ var _ = Describe("[rfe_id:273][crit:high][vendor:cnv-qe@redhat.com][level:compon
+					currentConfiguration.NetworkConfiguration = &v1.NetworkConfiguration{}
+				}
+
+				currentConfiguration.NetworkConfiguration.PermitBridgeInterfaceOnPodNetwork = pointer.BoolPtr(true)
+				kv := tests.UpdateKubeVirtConfigValueAndWait(currentConfiguration)
+				currentConfiguration = kv.Spec.Configuration
+
+				By("starting a VMI on a node")
+				vmi, err := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
+				Expect(err).To(BeNil(), "Should submit VMI successfully")
+
+				// Start a VirtualMachineInstance with bridged networking
+				nodeName := tests.WaitForSuccessfulVMIStart(vmi)
+
+				By("triggering a device plugin re-registration on that node")
+				pod, err := kubecli.NewVirtHandlerClient(virtClient).Namespace(flags.KubeVirtInstallNamespace).ForNode(nodeName).Pod()
+				Expect(err).ToNot(HaveOccurred())
+
+				_, _, err = tests.ExecuteCommandOnPodV2(virtClient, pod,
+					"virt-handler",
+					[]string{
+						"rm",
+						// We want to fail if the file does not exist, but don't want to be asked
+						// if we really want to remove write-protected files
+						"--interactive=never",
+						device_manager.SocketPath("kvm"),
+					})
+				Expect(err).ToNot(HaveOccurred())
+
+				By("checking if we see the device plugin restart in the logs")
+				virtHandlerPod, err := kubecli.NewVirtHandlerClient(virtClient).Namespace(flags.KubeVirtInstallNamespace).ForNode(nodeName).Pod()
+				Expect(err).ToNot(HaveOccurred(), "Should get virthandler client for node")
+
+				handlerName := virtHandlerPod.GetObjectMeta().GetName()
+				handlerNamespace := virtHandlerPod.GetObjectMeta().GetNamespace()
+				seconds := int64(10)
+				logsQuery := virtClient.CoreV1().Pods(handlerNamespace).GetLogs(handlerName, &k8sv1.PodLogOptions{SinceSeconds: &seconds, Container: "virt-handler"})
+				Eventually(func() string {
+					data, err := logsQuery.DoRaw()
+					Expect(err).ToNot(HaveOccurred(), "Should get logs")
+					return string(data)
+				}, 60, 1).Should(
+					ContainSubstring(
+						fmt.Sprintf("device socket file for device %s was removed, kubelet probably restarted.", "kvm"),
+					), "Should log device plugin restart")
+
+				// This is a little bit arbitrar
+				// Background is that new pods go into a crash loop if the devices are still report but virt-handler
+				// re-registers exactly during that moment. This is not too bad, since normally kubelet itself deletes
+				// the socket and knows that the devices are not there. However we have to wait in this test a little bit.
+				time.Sleep(10 * time.Second)
+
+				By("starting another VMI on the same node, to verify everything still works")
+				newVMI := newCirrosVMI()
+				newVMI.Spec.NodeSelector = map[string]string{"kubernetes.io/hostname": nodeName}
+				newVMI, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(newVMI)
+				Expect(err).To(BeNil())
+
+				tests.WaitForSuccessfulVMIStart(newVMI)
```

Comment:
doesn't the original vmi need to be verified as well? we're wanting to make sure it stays online after the kubelet restart, right?